File: src/backend/identityaccess/src/main/java/org/taonaw/identityaccess/domain/shared/exception/DomainException.java
package org.taonaw.identityaccess.domain.shared.exception;

import lombok.NonNull;

public class DomainException extends  RuntimeException {
    public DomainException(@NonNull DomainExceptionCodes code) {
        this(code, code.getDefaultMessage());
    }
    public DomainException(@NonNull DomainExceptionCodes code, String message) {
        super(code.getDefaultMessage());
    }
}

File: src/backend/identityaccess/src/main/kotlin/org/taonaw/identityaccess/domain/shared/exception/DomainException.kt
package org.taonaw.identityaccess.domain.shared.exception

class DomainException(message: String) : Exception(message) {
    private val errNotification: ErrNotification = ErrNotification()

    constructor(message: String, errNotification: ErrNotification) : this(message) {
        this.errNotification.addErr(errNotification)
    }

    fun throwIfHasErrs() {
        if (errNotification.hasErrs())
            throw this
    }

    override fun toString(): String {
        return super.toString() + "{DomainException(errNotification=${errNotification})}"
    }
}